hat/assets/js/apps/Iaso/domains/dataSources/components/AddNewEmptyVersion.js

```diff
@@ -0,0 +1,126 @@
+import React from 'react';
+import { Grid, Typography } from '@material-ui/core';
+import { FormattedMessage } from 'react-intl';
+import { LoadingSpinner, useSafeIntl } from 'bluesquare-components';
+import PropTypes from 'prop-types';
+import MESSAGES from '../messages';
+import ConfirmCancelDialogComponent from '../../../components/dialogs/ConfirmCancelDialogComponent';
+import { useCreateSourceVersion } from '../requests';
+import { useFormState } from '../../../hooks/form';
+import InputComponent from '../../../components/forms/InputComponent';
+
+const initialFormState = () => {
+    return {
+        versionDescription: '',
+    };
+};
+
+const VersionDescription = ({ formValue, onChangeDescription }) => {
+    const { formatMessage } = useSafeIntl();
+    return (
+        <InputComponent
+            type="text"
+            keyValue="versionDescription"
+            labelString={formatMessage(MESSAGES.dataSourceDescription)}
+            value={formValue}
+            onChange={(field, value) => {
+                onChangeDescription(field, value);
+            }}
+        />
+    );
+};
+
+const AddNewEmptyVersion = ({
+    renderTrigger,
+    sourceId,
+    forceRefreshParent,
+}) => {
+    // eslint-disable-next-line no-unused-vars
+    const [form, setFormField, , setFormState] = useFormState(
+        initialFormState(),
+    );
+    const { mutateAsync: createSourceVersion } = useCreateSourceVersion();
+
+    const reset = () => {
+        setFormState(initialFormState());
+    };
+
+    const submit = async closeDialogCallBack => {
+        const body = {
+            dataSourceId: sourceId,
+            description: form.versionDescription.value || null,
+        };
+        await createSourceVersion(body);
+        closeDialogCallBack();
+        reset();
+        forceRefreshParent();
+    };
+
+    const onConfirm = async closeDialog => {
+        await submit(closeDialog);
+    };
+
+    const onRedirect = async closeDialog => {
+        await submit(closeDialog);
+    };
```

Comment:
You can delete this, you use for `onAdditionalButtonClick` but since you just use the existing save and cancel buttons, it will never be called